contracts/Tokens/Prime/Prime.sol

```diff
@@ -217,6 +223,14 @@ contract Prime is IPrime, AccessControlledV8, PausableUpgradeable, MaxLoopsLimit
         _pause();
     }
 
+    /**
+     * @notice Prime initializer V2 for initializing pool registry
+     * @param poolRegistry_ Address of IL pool registry
+     */
+    function initializeV2(address poolRegistry_) external reinitializer(2) {
+        poolRegistry = poolRegistry_;
```

Comment:
What if we define `poolRegistry` as (public) immutable? We would avoid the re-initialization step. Moreover, I don't think we'll have to change it after deploying it, and if we have to do it we can always upgrade the implementation